### PR TITLE
Delete car from list

### PR DESCRIPTION
**Pull request by myelin0**
## In this branch I implement the deletion of car from the list.

- Add a destroy method to cars_controller.rb
- Add a route for deleting car item.
- Pass linters.
- Follow proper gitflow.

### CODE REVIEW

File: app/controllers/api/v1/cars_controller.rb [hunk 1]
```diff
@@ -25,6 +25,16 @@ def add_car
...
+  # delete a car item from list
+  def destroy
+    car = Car.find(params[:id])
+    if car.destroy
+      render json: { message: 'Car deleted successfully' }, status: :ok
+    else
+      render json: { head: unprocessable_entity }
+    end
+  end
+
```
Comment (nottherealalanturing) [inline app/controllers/api/v1/cars_controller.rb:37]: Nice job implementing the delete action.

**Review by nottherealalanturing — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi @myelin0  👋 , 

Great work implementing the delete action.



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good UI design
✔️ No linter errors

 ## Required Changes ♻️
_Check the comments under the review._

- [x]  Add the delete action to the before_action filter, to make it only accessible to authenticated users. 
 newaa



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nottherealalanturing — APPROVED** [review]
Hi @myelin0  😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉


Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/api/v1/cars_controller.rb (hunk 1 above)
Comment (myelin0) [reply app/controllers/api/v1/cars_controller.rb:37]: thanks! 😃

**Pull request merged**